### PR TITLE
Fixing VRF fluid control model TU cooling rate much larger than system 

### LINKED ISSUE #10187: VRF fluid control model cooling rate at terminal unit is much higher than cooling rate at system level
Issue overview
--------------
The terminal unit cooling rate is higher than the system cooling rate when the system cannot meet the load. The following output is produced when the "Rated Evaporative Capacity {W}" is set to 500W while "Rated Total Cooling Capacity {W}" is 10023 W. The "Cooling Coil Total Cooling Rate" is much larger than "VRF Heat Pump Total Cooling Rate".

image

With the following idf setting

image

### Defect file
[US+SF+CZ4A+hp+crawlspace+IECC_2006_VRF.idf.txt](https://github.com/NREL/EnergyPlus/files/12479284/US%2BSF%2BCZ4A%2Bhp%2Bcrawlspace%2BIECC_2006_VRF.idf.txt)

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by yujiex**
Pull request overview
---------------------
 - Fixes #10187

It is parallel to issue #10159

## issue diagnosis

For both the original and the fluid-control VRF model, terminal unit capacity should have been constrained by condenser capacity, where a maximum allowable terminal unit capacity is calculated in `LimitTUCapacity(.)` for each VRF system. The maximum allowable terminal unit capacity is stored in `state.dataHVACVarRefFlow->MaxCoolingCapacity(VRFCond)` and passed into `SimDXCoil(.)` which is NOT passed into `CalcVRFCoolingCoil_FluidTCtrl(.)`, as compared with the non-fluid-control VRF model, `CalcVRFCoolingCoil(.)`

This PR will add an optional argument `MaxCoolCap` to `CalcVRFCoolingCoil_FluidTCtrl(.)` as is in `CalcVRFCoolingCoil(.)` and use this quantity to constrain the cooling coil capacity.

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/DXCoils.cc [hunk 1]
```diff
@@ -16838,7 +16840,12 @@ void CalcVRFCoolingCoil_FluidTCtrl(EnergyPlusData &state,
             ShowFatalError(state, format("{} \"{}\" - Rated total cooling capacity is zero or less.", thisDXCoil.DXCoilType, thisDXCoil.Name));
         }
 
-        TotCap = thisDXCoil.RatedTotCap(Mode);
+        if (present(MaxCoolCap)) {
+            TotCap = min(MaxCoolCap, thisDXCoil.RatedTotCap(Mode));
+        } else {
+            TotCap = thisDXCoil.RatedTotCap(Mode);
+        }
+
```
Comment (rraustad) [inline src/EnergyPlus/DXCoils.cc:16848]: Could this be replaced as `TotCap = min(MaxCoolCap, thisDXCoil.RatedTotCap(Mode));` and just pass in the limit? or default it and only change this one line.

Comment (yujiex) [reply src/EnergyPlus/DXCoils.cc:16848]: Should the parent function `SimDXCoil` that calls `CalcVRFCoolingCoil_FluidTCtrl` also use a non-optional argument for MaxCap? Also the other parallel functions `CalcVRFCoolingCoil`, `CalcDXHeatingCoil`, `CalcVRFHeatingCoil_FluidTCtrl`? @rraustad

Comment (rraustad) [reply src/EnergyPlus/DXCoils.cc:16848]: I see, leave it as-is.

Comment (yujiex) [reply src/EnergyPlus/DXCoils.cc:16848]: Okay

**Comment by nrel-bot-2** [comment]
@yujiex @myoldmopar it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@yujiex @myoldmopar it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@yujiex @myoldmopar it has been 30 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@yujiex @myoldmopar it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@yujiex @myoldmopar it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@yujiex @myoldmopar it has been 28 days since this pull request was last updated.

**Pull request closed without merging**